### ISSUE TITLE
Mesh file format

### ISSUE THREAD

**Issue by BRiedel314** [nutofem/nuto#89]
According to issue #85 we have to specify a mesh file format containing all necessary information of a subdomain to assemble and solve a global system in parallel. Therefore each process reads one file to build its own local system. As a first approach (for isoparametric formulations) I would suggest a file in JSON format with the following content:
 - Nodes (local/global IDs, coordinates, master, slave)
 - Elements (type, global IDs, node IDs)
 - DOFs (type, global IDs, distinction master/slave/inner, distinction active/dependent, values)

As example see [here](https://gist.github.com/BRiedel314/3759cecd1693a9d627afbe460124320a).
Further questions are how to incorporate different DOF types (just straightforward?) and coupled constraints (e.g. u_1 = u_2).

**Comment by Psirus** [comment]
I don't understand why the "mesh" needs to save the DOF types, let alone their values. The nodes, elements, and maybe edges and boundaries, are related to the domain of the problem. But DOF values, and boundary conditions, etc. imho don't belong in a mesh file, they are part of the "physics" (at that point, you might as well serialize the `Structure`?!). I should be able to solve a temperature problem or a mechanics problem on the same domain, i.e. the same meshfile, if I wanted to, no?

And w.r.t. the file format, there a [a lot of serialization formats](https://en.wikipedia.org/wiki/Comparison_of_data_serialization_formats). JSON is nice because it's human-readable, but it's also large because of that. For size/speed, binary is king. For debugging purposes, it would be nice to have a format that can also be exported as ASCII. The [Protocal buffers](https://developers.google.com/protocol-buffers/docs/overview) are such a file format.

**Comment by joergfunger** [comment]
Mesh means the geometry interpolation as well as the interpolation for other dof types such as displacements or temperatures. For isoparametric formulations we do not need to make a difference. As soon as we use other interpolation types (e.g. linear polynomial for the geometry to simplify jacobians, but higher order displacements to increase accuracy) we have to convert the geometry mesh (nodes and elements) to the higher order displacement mesh. The idea discussed in #85 is to store the result of the conversion in a separate (mesh) file that includes the interpolation for all dof types. This also decouples (and simplifies) the problems of implementing trilinos domain decompositions (first decompose the domain with a consistens node numbering, master/slave etc, then solve.

The local to global map should be directly stored at the node (global id, local id).
The NodeToDof numbering should not be included in the mesh. Rather store the neighbor information between domains. There should also be the option to have multiple domains as neighbors.

**Comment by joergfunger** [comment]
See #92
